web_poet/page_inputs/http.py

```diff
@@ -300,3 +302,14 @@ def _auto_detect_fun(self, body: bytes) -> Optional[str]:
             except UnicodeError:
                 continue
             return resolve_encoding(enc)
+
+
+def request_fingerprint(req: HttpRequest) -> str:
```

Comment:
Let's add some basic tests for this function - e.g. that fingerprints of different requests are different, and fingerprints of similar (or equal) requests are the same.